app/grandchallenge/archives/models.py

```diff
@@ -13,4 +17,42 @@ class Archive(UUIDModel):
     images = models.ManyToManyField(Image)
 
     def __str__(self):
-        return "<{} {}>".format(self.__class__.__name__, self.name)
+        return f"<{self.__class__.__name__} {self.name}>"
+
+    def delete(self, *args, **kwargs):
+        """
+        Removes all patients, studies, images, imagefiles and annotations that belong
+        exclusively to this archive
+        """
+        images_to_remove = (
+            Image.objects.annotate(num_archives=Count("archive"))
+            .filter(archive__id=self.id, num_archives=1)
+            .all()
+        )
+        protected_study_ids = set()
+        protected_patient_ids = set()
+        for image in images_to_remove:
```

Comment:
I don't really understand what this block is trying to do, could you split this out into a separate (maybe protected) function so that developers can easier understand the flow? Or, even a comment would do.

Reply:
I moved this functionality into a separate method and tried to add a comment that explains what it does. Please let me know if the explanation is clear!